p_kit/core/p_circuit.py

```diff
@@ -1,10 +1,22 @@
 import numpy as np
+from p_kit.utils.deprecation import deprecated
 
+PCIRCUIT_EXAMPLE = """
+    @psl.pcircuit(n_pbits=3)
+    class MyCircuit:
+        input1 = Port('input1')
+        input2 = Port('input2')
+        output = Port('output')
+"""
 
+@deprecated("@psl.pcircuit decorator", example=PCIRCUIT_EXAMPLE)
 class PCircuit():
```

Comment:
Should we instead rewrite PCircuit to return a fixture that follows the decorator pattern?
This way, we leave it open to people that prefers composition.